solution/build.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+PWD=$(pwd)
+BUILD_ROOT=${PWD}/_build
+ISO_ROOT=${BUILD_ROOT}/root
+IMAGES_ROOT=${ISO_ROOT}/images
+
+PRODUCT_NAME=Zenko
+PRODUCT_LOWERNAME=zenko
+BUILD_TIMESTAMP=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+BUILD_HOST=$(hostname)
+
+VERSION_SHORT=$(git describe --abbrev=0)
+VERSION_FULL=${VERSION_SHORT}-dev
+GIT_REVISION=$(git describe --long --always --tags --dirty)
+ISO=${BUILD_ROOT}/${PRODUCT_LOWERNAME}-${VERSION_FULL}.iso
+
+DOCKER=docker
+DOCKER_OPTS=
+DOCKER_SOCKET=unix:///var/run/docker.sock
+HARDLINK=hardlink
+OPERATOR_SDK=operator-sdk
+OPERATOR_SDK_OPTS=
+SKOPEO=skopeo
+SKOPEO_OPTS="--override-os linux --insecure-policy"
+OPERATOR_TAG=$(grep /zenko-operator: deps.txt | awk -F ':' '{print $2}')
+
+# grab our dependencies from our deps.txt file as an array
+readarray -t DEP_IMAGES < deps.txt
+
+function clean()
+{
+	rm -rf ${BUILD_ROOT}
+}
+
+function mkdirs()
+{
+	mkdir -p ${ISO_ROOT}
+}
+
+function copy_yamls()
+{
+	sed "s/ZENKO_OPERATOR_TAG/${OPERATOR_TAG}/" config.yaml > ${ISO_ROOT}/config.yaml
+	cp -R -f operator/ ${ISO_ROOT}/operator
+	sed "s/VERSION_FULL/${VERSION_FULL}/" zenko-version-cr.yaml > ${ISO_ROOT}/operator/zenko-version-cr.yaml
+}
+
+function download_dependency_yamls()
+{
+	OPERATOR_DIR=${ISO_ROOT}/operator
+	CRDS_DIR=${OPERATOR_DIR}/crds
+
+	wget --directory=${CRDS_DIR} https://github.com/jetstack/cert-manager/releases/download/v0.11.0/cert-manager.yaml || exit 1
```

Comment:
@tcarmet I added some `|| exit 1` and file checks to make bad stuff actually exit properly.

Reply:
`set -e` at the top of the script will make anything that fails exit. It's generally good practice along with `set -u`.

Reply:
hmm, gotta look up -u. I tried `-e` and it made some other things fail when they were fine. But will give it another shot. I think it was my extra if checks were causing issues that probably just need to not exist with `-e`.

Reply:
`-u` (combined with `-e`) aborts when reading an undefined variable. `-e` can cause problems on scripts that were written without it in the first place, but any well-written script will generally work fine with `-e`.

Reply:
^ sweet yea, that works now. I was doing extra if checks which kinda voided the -e before. Removed those ( after checking they still work ) and added the -u.